### PR TITLE
2440: handle empty appointments responses

### PR DESCRIPTION
**Pull request by kpethtel**
## Description of change
Fixes an issue found in staging in which empty appointments responses cause a syntax error. 
http://sentry.vfs.va.gov/organizations/vsp/issues/49344/events/latest/?project=3&query=is%3Aunresolved

## Original issue(s)
department-of-veterans-affairs/va-mobile-team#2440

### CODE REVIEW

File: modules/mobile/spec/request/appointments_list_request_spec.rb [hunk 1]
```diff
@@ -890,5 +890,43 @@
...
+    context 'when no va appointments are returned' do
+      before do
+        Timecop.freeze(Time.zone.parse('2020-11-01T10:30:00Z'))
+      end
+
+      after { Timecop.return }
+
+      it 'returns 200' do
+        VCR.use_cassette('appointments/get_cc_appointments', match_requests_on: %i[method uri]) do
+          VCR.use_cassette('appointments/get_appointments_empty', match_requests_on: %i[method uri]) do
```
Comment (kpethtel) [inline modules/mobile/spec/request/appointments_list_request_spec.rb:903]: These empty vcr cassettes were taken from the vaos cassette collection.

File: modules/mobile/app/services/mobile/v0/appointments/proxy.rb [hunk 2]
```diff
@@ -59,8 +59,10 @@ def legacy_fetch_appointments(start_date, end_date)
           va_appointments = va_appointments_adapter.parse(va_response[:response].body)
           cc_appointments = cc_appointments_adapter.parse(cc_response[:response].body)
 
-          facilities = fetch_facilities(va_appointments)
-          va_appointments = backfill_appointments_with_facilities(va_appointments, facilities)
+          if va_appointments.any?
+            facilities = fetch_facilities(va_appointments)
+            va_appointments = backfill_appointments_with_facilities(va_appointments, facilities)
```
Comment (kpethtel) [inline modules/mobile/app/services/mobile/v0/appointments/proxy.rb:64]: It's still plausible that errors could occur with facilities, but I think that was a pre-existing possibility and my work didn't change it.

**Review by jperk51 — APPROVED** [review]
lgtm

**Review by aherzberg — APPROVED** [review]
LGTM

**Pull request merged**